src/oracle/middleware.js

```diff
@@ -0,0 +1,761 @@
+
+var credentials;
+var feeds;
+
+/**
+  * Function to initialise state
+  */
+const initialiseState = async () => {
+  //create tables if they do not exist
+  await createDBs();
+}
+
+/**
+  * Function to send a job run to the CL node
+  * @param {*} credentials the external initiator credentials
+  * @param {*} count the request id
+  * @param {*} jobId the Chainlink external job id
+  * @param {*} chainlinkUrl the Chainlink node url where to send the job request
+  * @param {*} requestType the request type, 1 = time, 2 = deviation, 3 = new round
+  */
+const sendJobRun = async (credentials, count, jobId, chainlinkUrl, requestType) => {
+  const options = {
+    url: chainlinkUrl + "/v2/jobs/" + jobId + "/runs",
+    body: {
+      "payment": 0,
+      "request_id": count,
+      "request_type": requestType
+    },
+    headers: {
+      "Content-Type": "application/json",
+      "X-Chainlink-EA-AccessKey": credentials["EI_IC_ACCESSKEY"],
+      "X-Chainlink-EA-Secret": credentials["EI_IC_SECRET"]
+    },
+    method: 'POST',
+  };
+
+  //try request with loop retries
+  for (let i = 0; i < SUBMIT_RETRIES; i++) {
+    try {
+      await axios.post(options.url, options.body, {
+        timeout: 60000,
+        proxy: false,
+        headers: options.headers,
+        httpAgent: new http.Agent({ keepAlive: false })
+      });
+      return
+    }
+    catch (err) {
+      console.error("JOB Request for " + jobId + " failed", err)
+    }
+  }
+}
+
+/**
+  * Function to get last 10 offers
+  * @param {*} follower offers and balances
+  * @returns a list of offers
+  */
+export const getOffers = async (follower) => {
+
+  let history = [];
+  let counter = 0;
+  let lastVisited = 0;
+
+  for await (const followerElement of iterateReverse(follower)) {
+
+    if (counter == 5){
+      break;
+    }
+
+    //if it is an offer status 
+    if (followerElement.value.updated == "offerStatus"){
+      //get id
+      let id = followerElement.value.status.id
+
+      //if a new and final state
+      if (id != lastVisited) {
+        //if it is not failed
+        if (!followerElement.value.status.hasOwnProperty("error")) {
+          history.push(followerElement.value);
+          counter++
+        }
+        lastVisited = id;
+      } 
+    }
+  }
+  return history
+}
+
+/**
+ * Function to get the latest submitted round
+ * @returns the latest round submitted
+ */
+export const getLatestSubmittedRound = async () => {
+  const unserializer = boardSlottingMarshaller(fromBoard.convertSlotToVal);
+  const leader = makeLeader(networkConfig.rpcAddrs[0]);
+
+  const follower = await makeFollower(
+    `:published.wallet.${FROM}`,
+    leader,
+    {
+      // @ts-expect-error xxx
+      unserializer,
+    },
+  );
+
+  //get offers
+  let offers = await getOffers(follower)
+
+  return Number(offers[0]["status"]["invitationSpec"]["invitationArgs"][0]["roundId"])
+}
+
+/**
+ * Function to check if submission was satisfied for a specific round
```

Comment:
What do you mean submission was satisfied? seems to imply that a submission was submission successful rather than a submission needs to be submitted?

Reply:
Fixed in https://github.com/jacquesvcritien/agoric-cl-middleware/commit/de8807b1ff7ee9fdc5fe1fcd3c29dd9899b5a0df